packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md

```diff
@@ -0,0 +1,116 @@
+> **Note**: The examples in this guide use Java 11 and Spring Boot MVC
+
+### 1: Start password recovery
+
+After the user starts the password recovery flow and selects the email authenticator for the process, Okta sends them an email matching the **Forgot Password** template you altered earlier.
+
+<div class="common-image-format">
+
+![Example of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password recovery email")
+
+</div>
+
+When the user clicks the **Reset Password** link, their browser sends a request to the endpoint defined by the template, attaching the `${oneTimePassword}` and `${request.relayState}` VTL variables as query parameters to the URL. For instance, in the sample this request might render as `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler that takes the `${oneTimePassword}` and `${request.relayState}` values from the query string, saves them into local variables (for instance, 'otp', and 'state'), and makes the following checks:
+
+1. That the current session attribute `IdxClientContext` is not `null` (because the user clicked the magic link from a different browser).
```

Comment:
```suggestion
1. The current session attribute `IdxClientContext` isn't `null` (because the user clicked the magic link from a different browser).
```

Reply:
Changed isn't, but didn't lose "That" for the same reason as above.